### PR TITLE
fix: enforce scheduling registry pod onto infra nodes

### PR DESCRIPTION
**Pull request by anispate**
This commit specifies .spec.grpcPodConfig.nodeSelector and .spec.grpdPodConfig.tolerations to schedule this operator's registry pod to infra nodes when possible.

[OSD-6629](https://issues.redhat.com//browse/OSD-6629)

### CODE REVIEW

File: hack/olm/catalog-src.yaml
Comment (sthaha) [inline hack/olm/catalog-src.yaml:20]: @anispate , the PR says `prefer` instead of enforce. Is that really the case? I haven't tested this but my hunch is that this enforces Catalog pods to be scheduled only on infra nodes, doesn't it?

Comment (anispate) [reply hack/olm/catalog-src.yaml:20]: It was already tested before by us for other operators.

Comment (sthaha) [reply hack/olm/catalog-src.yaml:20]: If it enforces, like mentioned in the slack thread, we wouldn't want to enforce this on all users of the catalog. 
cc: @jan--f

Comment (mjlshen) [reply hack/olm/catalog-src.yaml:20]: Agreed, this configuration enforces that it can only be scheduled onto infra nodes. We likely need a separate CatalogSource for OSD/ROSA compared to the generic one

**Review by cblecker — APPROVED** [review]
This only updates the SelectorSyncSet used for managed deployments

File: deploy/olm/syncselector-template.yaml
Comment (sthaha) [inline deploy/olm/syncselector-template.yaml:1]: @jan--f are we okay with this change? I feel it is fine since it only modifies the syncselector-template.

**Review by jan--f — CHANGES_REQUESTED** [review]
Sorry for blocking this, I think we should have an ack from the SD org.
For the deployment of the operator we were specifically asked to prefer (not require) infra nodes. Like in https://github.com/rhobs/observability-operator/pull/313.

Any reason we can't do the same for the CatalogSource?

**Comment by jan--f** [comment]
2023-07-25-155005_852x72

The commit needs this name change :)

**Comment by apahim** [comment]
The reason SD previously reccomended no "prefer" instead of "enfonce" was because we observed some cases where infra nodes didn't have capacity or were not available to take the workloads, but those were isolated cases.

The current understanding is that we are fine with enforcing the scheduling of infra bits to infra nodes, as SREP will receive an alert saying "InfraNodesNeedResizingSRE" and resize the nodes when needed.

This PR looks good to me.

**Review by jan--f — APPROVED** [review]
Thanks @apahim!
This lgtm.

**Comment by sthaha** [comment]
@jan--f before merging, should the `prefer` in commit message be changed to `enforce` ?

**Pull request merged**
